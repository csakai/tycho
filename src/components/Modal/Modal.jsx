import React from 'react';
import PropTypes from 'prop-types';
import ReactMarkdown from 'react-markdown';
import cx from 'classnames';
import './Modal.css';

export default class Modal extends React.Component {

  static propTypes = {
    modalActive: PropTypes.bool,
    closeModal: PropTypes.func.isRequired,
    title: PropTypes.string,
    description: PropTypes.string.isRequired,
    velocity: PropTypes.number,
    magnitude: PropTypes.number
  }

  render() {
    return (
      <div className={`modal modal--${cx({
        open: this.props.modalActive,
        closed: !this.props.modalActive 
      })}`}>
        <div className="modal__header">
          <span>{this.props.title}</span>
					<span className="modal__close" onClick={this.props.closeModal}>
						&times;
					</span>
				</div>
				
				<div className="modal__content">
          <ReactMarkdown source={this.props.description} />
        </div>
				
        <div className="modal__footer">
          <div className="modal__info">
            <span>Velocity at vector:</span><br />
            <span>Distance to Sun:</span>
          </div>
          <div className="modal__info">
            <span>{this.props.velocity} km/s</span><br />
            <span>{this.props.magnitude} AU</span>
          </div>
				</div>
			</div>
    );
  }
}
