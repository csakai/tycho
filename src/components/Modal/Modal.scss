@import 'globals.scss';

$min-width: 600px;

.modal {
    line-height: $font-line-height;
    background-color: $opaque-backdrop;
    color: $brand-primary;
    display: flex;
    flex-direction: column;
    position: absolute;
    overflow: hidden;
    box-sizing: border-box;
    top: 0;
    right: -999px;
    bottom: 0;
    width: 40%;
    min-width: $min-width;
    opacity: 0;
    padding: $base-spacing;
    transition-property: background-color, opacity, right;
    transition-duration: $transition-speed;
    z-index: $z-index-top;

    @media screen and (max-width: $min-width) {
        width: 100%;
        min-width: 100%;
    }

    &--open {
        right: 0;
        opacity: 1;
    }

    &__header {
        border-bottom: $dotted-border;
        font-size: $font-size-large;
        padding: ($base-spacing / 2) 0;
    }

    &__content {
        display: flex;
        font-size: $font-size-tiny;
        line-height: $font-line-height;

        p {
            text-indent: 50px;
        }

        a {
            color: $brand-primary;
            &:hover {
                text-decoration: none;
            }
        }
    }

    &__close {
        float: right;
        cursor: pointer;
        transition: all $transition-speed;
        &:hover {
            text-shadow: $text-shadow;
        }
    }
}
