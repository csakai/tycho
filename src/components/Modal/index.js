import './Modal.css';
export {default} from './Modal';
