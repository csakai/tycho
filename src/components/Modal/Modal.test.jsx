import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import Modal from './';

describe('Modal Component', () => {
    let component;

    beforeEach(() => {
        component = shallow(
            <Modal
                closeModal={jest.fn()}
                title="Test Modal"
                description="Test Description"
            />
        );
    });

    it('should render the Modal successfully', () => {
        expect(toJson(component)).toMatchSnapshot();
    });
});
