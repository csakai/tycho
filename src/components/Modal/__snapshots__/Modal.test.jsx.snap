// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component should render the Modal successfully 1`] = `
<div
  className="modal modal--closed"
>
  <div
    className="modal__header"
  >
    <span>
      Test Modal
    </span>
    <span
      className="modal__close"
      onClick={[Function]}
    >
      Ã—
    </span>
  </div>
  <div
    className="modal__content"
  />
</div>
`;
